From: Ben Hutchings <ben@decadent.org.uk>
Subject: pps: avoid ABI change in 3.2.40
Date: Wed, 06 Mar 2013 14:15:34 +0000

Move the new member pps_device::lookup_cookie to the end of the
structure and hide it from genksyms.  This structure is always
allocated by pps_register_source().

---
--- a/include/linux/pps_kernel.h
+++ b/include/linux/pps_kernel.h
@@ -69,11 +69,14 @@ struct pps_device {
 	wait_queue_head_t queue;		/* PPS event queue */
 
 	unsigned int id;			/* PPS source unique ID */
-	void const *lookup_cookie;		/* pps_lookup_dev only */
 	struct cdev cdev;
 	struct device *dev;
 	struct fasync_struct *async_queue;	/* fasync method */
 	spinlock_t lock;
+
+#ifndef __GENKSYMS__
+	void const *lookup_cookie;		/* pps_lookup_dev only */
+#endif
 };
 
 /*
